src/action-emitter.js

```diff
@@ -33,10 +33,18 @@ export function makeActionEmitter() {
             // this means that if an action is taken by two sagas and one of those
             // sagas emits an action taken by the 2nd saga, it will not pick it up
             // since it will process the previous action *afterwards*
+            let firstThrown;
             for (let i = 0; i < listenersToFire.length; i++) {
 
                 // callback follow redux-tale callback format isThrown, value
-                listenersToFire[i].callback(false, action);
+                try {
+                    listenersToFire[i].callback(false, action);
+                } catch (e) {
+                    firstThrown = firstThrown || e;
+                }
+            }
+            if (firstThrown) {
+                throw firstThrown;

```

Comment:
I don't like that this can lose exceptions.
Do we really need to re-throw at all? or should they all be unhandled and logged as per unhandled errors?

Reply:
Pulled the error logging out into separate file for use here and in effects.js